File: library/src/main/java/com/dylanc/retrofit/helper/RetrofitManager.java
package com.dylanc.retrofit.helper;

import me.jessyan.progressmanager.ProgressListener;
import me.jessyan.progressmanager.ProgressManager;
import me.jessyan.retrofiturlmanager.RetrofitUrlManager;

public class RetrofitManager {

  public static void setGlobalDomain(String globalDomain) {
    RetrofitUrlManager.getInstance().setGlobalDomain(globalDomain);
  }

  public static void putDomain(String domainName, String domainUrl) {
    RetrofitUrlManager.getInstance().putDomain(domainName, domainUrl);
  }

  public static void addRequestListener(String url, ProgressListener listener){
    ProgressManager.getInstance().addRequestListener(url,listener);
  }
}

File: library/src/main/java/com/dylanc/retrofit/helper/RetrofitManager.kt
@file:Suppress("unused")

package com.dylanc.retrofit.helper

import com.franmontiel.persistentcookiejar.PersistentCookieJar
import me.jessyan.retrofiturlmanager.RetrofitUrlManager

/**
 * @author Dylan Cai
 * @since 2019/7/13
 */
object RetrofitManager {

  @JvmStatic
  fun putDomain(domainName: String, domainUrl: String) {
    RetrofitUrlManager.getInstance().putDomain(domainName, domainUrl)
  }

  @JvmStatic
  fun setGlobalDomain(globalDomain: String) {
    RetrofitUrlManager.getInstance().setGlobalDomain(globalDomain)
  }

  @JvmStatic
  fun clearCookieJar() {
    val cookieJar = RetrofitHelper.default.cookieJar
    if (cookieJar is PersistentCookieJar) {
      cookieJar.clear()
    }
  }
}
